app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsFragment.kt

```diff
@@ -366,24 +358,19 @@ class NotificationsFragment :
                 }
 
                 // Update status display from statusDisplayOptions. If the new options request
-                // relative time display collect the flow to periodically re-bind the UI.
+                // relative time display collect the flow to periodically update the timestamp in the list gui elements.
                 launch {
                     viewModel.statusDisplayOptions
                         .collectLatest {
+                            // NOTE this this also triggered (emitted?) on resume.
+
                             adapter.statusDisplayOptions = it
-                            layoutManager.findFirstVisibleItemPosition().let { first ->
-                                first == RecyclerView.NO_POSITION && return@let
-                                val count = layoutManager.findLastVisibleItemPosition() - first
-                                adapter.notifyItemRangeChanged(
-                                    first,
-                                    count,
-                                    null
-                                )
-                            }
+                            adapter.notifyItemRangeChanged(0, adapter.itemCount, null)
 
                             if (!it.useAbsoluteTime) {
                                 updateTimestampFlow.collect()
                             }
+                            // TODO there is a way to stop this?
```

Comment:
It stops automatically. 

https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/collect-latest.html

"The crucial difference from collect is that when the original flow emits a new value then the action block for the previous value is cancelled."

So when there's a new value for statusdisplayoptions the previous update job (if it exists) is cancelled automatically.